File: LeetCode/LeetCode Practice Problems/1725- Number Of Rectangles That Can Form The Largest Square.cpp
class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        map<int,int> m;
        int mx = 0;
        
        for(auto i :rectangles) {
            int mn = min(i[0] , i[1]);
            m[mn]++;
            mx = max(mx , mn);
        }
        
        return m[mx];
    }
};

File: LeetCode/LeetCode Practice Problems/1725-Number Of Rectangles That Can Form The Largest Square.java
class Solution {
    public int countGoodRectangles(int[][] rectangles) {
        int mx = 0;
        int cnt = 0;
        
        for(int rectangle[]:rectangles) {
            int mn = Math.min(rectangle[0], rectangle[1]);
            
            if(mx < mn) {
                mx = mn;
                cnt = 0;
            }
            
            if(mx == mn) {
                cnt++;
            }
        }
        
        return cnt;
    }
}
